### ISSUE TITLE
FR: Limit Navigation Output of FE-Module to certain RootPages

### ISSUE THREAD

**Issue by mvscheidt** [terminal42/contao-changelanguage#135]
**This is less of a feature request and more a discussion / checking if you are interested in a PR for a feature we needed anyways**: Limiting the FE-Module Output to configurable root pages.

minimal example root page structure

* `en` -> `foobar.com`
* `de-DE` -> `foobar.com/de-DE/`
* `de-CH` -> `foobar.com/de-CH/`

**Goal**: On `de-DE` I want to prevent output of `de-CH` in the Language Navigation completely (no leftovers in the markup either).

I know you can technically already do this via css / template. In my opinion both of these are not a very good solutions because for both you load and process data needlessly. In case of a css-based solution it's also still part of the markup which is also not optimal.

I'll just give two screenshots and try to explain how it works currently (in our branch) - please bear with the fact that there are not translations currently (mainly because if you want a PR I need to know how I should prepare the translations / xlf files anyways)

The first checkbox is the "on/off" switch for this feature, obviously.

----

![only-de-de-and-en](https://user-images.githubusercontent.com/6872430/29873897-42b19d8a-8d95-11e7-8eef-b21a2d43b368.png)

This config means that when this module gets generated in the frontend it will only include pages whose rootIds are defined here.

Thus the FE-Module Output will be 

* `en`
* `de-DE`

----

![all-but-de-ch](https://user-images.githubusercontent.com/6872430/29873896-429041a8-8d95-11e7-9461-643fbe1dfca4.png)

This one is similar to the first, except that the last checkbox is active which simply negates the condition used.

This means that when this module gets generated in the frontend it will include all pages EXCEPT the ones defined here.

In other words its the same as the case above

* `en`
* `de-DE`

The last checkbox is just for convenience based on the current project needs / future plans - in our case we will use it, because we have way more entries which should appear in the language navigation than entries which should not.

----

Lastly, there is one edge case (which is handled already): Depending on the last checkboxes state there is a need to **add** (when checkbox is unchecked) or **remove** (when its checked) the rootId of the current page to the config (dynamically - the module config is not changed) - otherwise you might not see the "active" language in some configurations.

I also added some tests for this where applicable - the old tests ran fine.

**Comment by aschempp** [comment]
Can you give an example why such a configuration would be needed?

**Comment by mvscheidt** [comment]
For example - you use Geo-IP to prevent people within Country A (on Page A) to visited a page made for Country B (Page B). If people of Country A try to enter Page B they get send to Page A instead. Due to law basically if I understood it correctly.

In such a case for people within Country A (on Page A) it would be useless to be able to switch to Page B (because they would get send back).

----

So it's quite an edge case, but before the feature is withering away in our fork I'd rather ask if you want it or not.

**Comment by aschempp** [comment]
Well as you say, it's rather an edge case. The extension exists for 10 years now and that issue never came up. It also does not work without other extensions (GeoIP detection), so I feel like it does not make sense in the core distribution.

However, regarding your implementation, you might want to look at the provided hook/event to remove items from the menu. That might be a better option than in the template.

**Comment by mvscheidt** [comment]
Fair enough :)

Yeah the Hook would indeed be better (overlooked it) - especially because I can do this from another module and save myself the hassle of keeping our fork up to date with the core distro while preserving our additions. Still need to think about some things but using the hook seems preferable to me.

I'll close this then - thanks for your time
